docs: clarify that `matcher` needs to be constant values (#38597)
vercel/next.js: docs/advanced-features/middleware.md

}
```

> > **Note:** The `matcher` values need to be constants so they can be analyzed at build-time. Non-static values will be ignored.

Reviewer: ```suggestion
> **Note:** The `matcher` values need to be constants so they can be statically analyzed at build-time. Dynamic values such as variables will be ignored.
```
